Add MQTT BBEs (#4829)
ballerina-platform/ballerina-distribution: examples/mqtt-client-basic-authentication/mqtt_client_basic_authentication.md

# MQTT client - Basic authentication

> The `mqtt:Client` connects to an MQTT server via basic authentication and then, sends messages to the server. Basic authentication can be enabled by configuring the `username`, `password` in the `connectionConfig` of the `mqtt:ClientConfiguration`. Use this to connect to an MQTT server secured with basic authentication.

Reviewer: ```suggestion
The `mqtt:Client` connects to an MQTT server via basic authentication and then, sends messages to the server. Basic authentication can be enabled by configuring the `username` and `password` in the `connectionConfig` of the `mqtt:ClientConfiguration`. This can be used to connect to an MQTT server secured with basic authentication.
```

> The `mqtt:Client` connects to a MQTT server via basic authentication and then, sends messages to the server. Basic authentication can be enabled by configuring the `username`, `password` in the `connectionConfig` of the `mqtt:ClientConfiguration`. Use this to connect to a MQTT server secured with basic authentication. Use this to connect to a MQTT server secured with basic authentication.

Author: Updated

::: code mqtt_client_basic_authentication.bal :::

> ## Prerequisites

Reviewer: ```suggestion
## Prerequisites
```
Shall we have these steps as numbered bullet points?

Author: This is the format other stdlib bbes are in.

> - Start a [MQTT broker](https://mqtt.org/software/) instance configured to use basic authentication.

Reviewer: ```suggestion
- Start an [MQTT broker](https://mqtt.org/software/) instance, which is configured to use basic authentication.
```
